packages/react/src/components/cookieConsentCore/cookieConsentCore.js

```diff
@@ -0,0 +1,670 @@
+   * @return {Promise<CookieConsentCore>} A promise that resolves to a new instance of the CookieConsent class.
+   * @throws {Error} Throws an error if the siteSettingsParam is not a string or an object.
+   * @throws {Error} Throws an error if siteSettingsParam is an URL string and the fetch fails.
+   * @throws {Error} Throws an error if siteSettingsParam is an URL string and the JSON parsing fails.
+   */
+  static async create(siteSettingsParam, options) {
+    let instance;
+    if (!siteSettingsParam && (typeof siteSettingsParam !== 'string' || typeof siteSettingsParam !== 'object')) {
+      throw new Error(
+        'Cookie consent: siteSettingsParam is required, it should be an URL string or an siteSettings object.',
+      );
+    }
+    if (typeof siteSettingsParam === 'string') {
+      // Fetch the site settings JSON file
+      const siteSettingsRaw = await fetch(siteSettingsParam).then((response) => {
+        if (!response.ok) {
+          throw new Error(
+            `Cookie consent: Unable to fetch cookie consent settings: '${response.status}' from: '${siteSettingsParam}' `,
+          );
+        }
+        return response.text();
+      });
+
+      // Parse the fetched site settings string to JSON
+      let siteSettingsObj;
+      try {
+        siteSettingsObj = JSON.parse(siteSettingsRaw);
+      } catch (error) {
+        throw new Error(`Cookie consent: siteSettings JSON parsing failed: ${error}`);
+      }
+      instance = new CookieConsentCore(siteSettingsObj, options, privateSymbol);
+    } else {
+      instance = new CookieConsentCore(siteSettingsParam, options, privateSymbol);
+    }
+
+    // Initialise the class instance
+    await instance.#init();
+
+    // Return reference to the class instance
+    return instance;
+  }
+
+  /**
+   * Get the consent status for the specified cookie group names.
+   * @param {string[]} groupNamesArray - An array of group names.
+   * @return {Promise<boolean>} A promise that resolves to true if all the groups are accepted, otherwise false.
+   */
+  getConsentStatus(groupNamesArray) {
+    return this.#COOKIE_HANDLER.getConsentStatus(groupNamesArray);
+  }
+
+  /**
+   * Sets the status of given cookie groups to accepted.
+   *
+   * @param {Array} acceptedGroupsArray - An array of cookie group names to be set as accepted.
+   * @return {Promise<boolean>} - A promise that resolves to true if the groups' status is successfully set to accepted, otherwise false.
+   */
+  async setGroupsStatusToAccepted(acceptedGroupsArray) {
+    return this.#COOKIE_HANDLER.setGroupsStatusToAccepted(acceptedGroupsArray);
+  }
+
+  /**
+   * Retrieves the status of the component.
+   * @return {Promise<{ cookie: string, monitor: string }>} The status object containing the cookie and monitor status.
+   */
+  async getStatus() {
```

Comment:
Nothing seems to use this. Same with handler.getStatus and monitor.getStatus

Reply:
It and corresponding other getStatus development only functions were removed.